pika/adapters/blocking_connection.py

```diff
@@ -1172,6 +1172,15 @@ def __int__(self):
     def __repr__(self):
         return '<%s impl=%r>' % (self.__class__.__name__, self._impl)
 
+    def __enter__(self):
+        return self
+
+    def __exit__(self, exc_type, value, traceback):
+        try:
+            self.close()
+        except (exceptions.ChannelClosed, exceptions.ChannelAlreadyClosing):
```

Comment:
It may be confusing/misleading to have something here that's not supposed to happen. Error-handling code, like other code, often gets copied, and unnecessary stuff (if indeed unnecessary) tends to spread around and add to confusion.